pymeasure/instruments/hp/hp34401A.py

```diff
@@ -298,24 +296,37 @@ def beep(self):
         """Control whether the beeper is enabled.""",
         validator=strict_discrete_set,
         values=BOOL_MAPPINGS,
-        map_values=True)
+        map_values=True
+    )
 
     scpi_version = Instrument.measurement(
         "SYST:VERS?",
-        """The SCPI version of the multimeter.""")
+        """The SCPI version of the multimeter."""
+    )
 
     stored_readings_count = Instrument.measurement(
         "DATA:POIN?",
-        """The number of readings currently stored in the multimeter's internal memory.""")
+        """The number of readings currently stored in the multimeter's internal memory."""
+    )
 
     self_test_result = Instrument.measurement(
         "*TST?",
         """Initiate a self-test of the multimeter and return the result.
 
         Be sure to set an appropriate connection timeout,
-        otherwise the command will fail.""")
-
-    def _get_function_range_prefix(self):
+        otherwise the command will fail."""
+    )
+
+    def write(self, command):
+        """Write a command to the instrument."""
+        if "<function_prefix_for_range>" in command:
+            command = command.replace("<function_prefix_for_range>",
+                                      self._get_function_prefix_for_range())
+        elif "<function>" in command:
+            command = command.replace("<function>", HP34401A.FUNCTIONS[self.function_])
+        super().write(command)
```

Comment:
As a suggestion: You could use the format language with curly braces (you have to change your occurences of `<abc>` to `{abc}`) as they indicate normally that you will format something in a string:
```suggestion
        if "{function_prefix_for_range}" in command:
            command = command.format(function_prefix_for_range=self._get_function_prefix_for_range())
        elif "{function}" in command:
            command = command.format(function=HP34401A.FUNCTIONS[self.function_])
        super().write(command)
```